src/macos/aarch64/vcpu.rs

```diff
@@ -58,15 +59,17 @@ impl VirtualizationBackendInternal for XhyveVm {
 
 	fn new(
 		peripherals: Arc<VmPeripherals>,
-		_params: &Params,
+		params: &Params,
 		guest_addr: GuestPhysAddr,
 	) -> HypervisorResult<Self> {
 		trace!("Create VM...");
 		create_vm()?;
 
 		trace!("Map guest memory...");
 		map_mem(
-			unsafe { peripherals.mem.as_slice_mut() },
+			unsafe {
```

Comment:
I think we should copy the `params.memory_size` into `peripherals` instead of occasionally (regularly?) having to stitch values from completely different places together.

Reply:
It'd probably even be better to incorporate the logic of multiple memory regions in here. This is already part of the type logic in the vm-memory crate.